presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -984,6 +1000,29 @@ public ConnectorTableHandle createTemporaryTable(ConnectorSession session, List<
         return new HiveTableHandle(schemaName, tableName);
     }
 
+    private Set<ColumnStatisticType> getSupportedColumnStatisticsForTemporaryTable(Type type)
+    {
+        // Temporary table statistics are not committed to metastore, so no need to call metastore for supported
+        // column statistics, instead locally determine.
+        if (type.equals(BOOLEAN)) {
+            return ImmutableSet.of(NUMBER_OF_NON_NULL_VALUES, NUMBER_OF_TRUE_VALUES);
+        }
+        if (isNumericType(type) || type.equals(DATE) || type.equals(TIMESTAMP)) {
+            return ImmutableSet.of(MIN_VALUE, MAX_VALUE, NUMBER_OF_DISTINCT_VALUES, NUMBER_OF_NON_NULL_VALUES);
+        }
+        if (isVarcharType(type) || isCharType(type)) {
+            return ImmutableSet.of(NUMBER_OF_NON_NULL_VALUES, NUMBER_OF_DISTINCT_VALUES, TOTAL_SIZE_IN_BYTES, MAX_VALUE_SIZE_IN_BYTES);
+        }
+        if (type.equals(VARBINARY)) {
+            return ImmutableSet.of(NUMBER_OF_NON_NULL_VALUES, TOTAL_SIZE_IN_BYTES, MAX_VALUE_SIZE_IN_BYTES);
+        }
+        if (type instanceof ArrayType || type instanceof RowType || type instanceof MapType) {
+            return ImmutableSet.of(NUMBER_OF_NON_NULL_VALUES, TOTAL_SIZE_IN_BYTES);
+        }
+        // For other unknown column types, specify default column statistics TOTAL_SIZE and NON_NULL_VALUES which is not specific to any data type.
```

Comment:
nit  `// default`    is enough 

or keep names same using TOTAL_SIZE_IN_BYTES and NUMBER_OF_NON_NULL_VALUES